extensions/wikia/Captcha/CaptchaController.class.php

```diff
@@ -5,17 +5,17 @@
  */
 class CaptchaController extends WikiaController {
 
-	const DEFAULT_TEMPLATE_ENGINE = WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
-
 	/**
 	 * Displays a captcha image. This is used exclusively by FancyCaptcha.
 	 */
 	public function showImage() {
 		$fancyCaptcha = new \Captcha\Module\FancyCaptcha();
-		if ( !$fancyCaptcha->showImage() ) {
-			$this->response->setData( [
-				'error' => wfMessage( 'captcha-no-image' )->escaped(),
-			] );
+		if ( $fancyCaptcha->showImage() ) {
+			/*
+			* Stop immediately when captcha is sent to browser.
+			* We don't want to send additional headers and other nirvana stuff
+			*/
+			exit();
```

Comment:
I don't think there are good grounds for exit like that.
I know that this is quick solution for PHP Warning we currently have, but we should at least create a new ticket for fixing/changing the controller to be able not sending headers when it should not.